reference/forms/types/choice.rst

```diff
@@ -85,19 +111,57 @@ choices
 
 This is the most basic way to specify the choices that should be used
 by this field. The ``choices`` option is an array, where the array key
-is the item value and the array value is the item's label::
+is the item's label and the array value is the item's value::
 
     $builder->add('gender', 'choice', array(
-        'choices' => array('m' => 'Male', 'f' => 'Female'),
+        'choices' => array('Male' => 'm', 'Female' => 'f'),
+        'choices_as_values' => true,
     ));
 
-.. tip::
+.. versionadded:: 2.7
+    Symfony 2.7 introduced the option to flip the ``choices`` array to be
+    able to use values that are not integers or strings (but e.g. floats
+    or booleans).
+
+choices_as_values
+~~~~~~~~~~~~~~~~~
 
-    When the values to choose from are not integers or strings (but e.g.
-    floats or booleans), you should use the `choice_list`_ option instead.
-    With this option you are able to keep the original data format which
-    is important to ensure that the user input is validated properly and
-    useless database updates caused by a data type mismatch are avoided.
+**type**: ``boolean`` **default**: false
+
+.. versionadded:: 2.7
+
+    The ``choices_as_values`` option of ChoiceType was introduced in Symfony 2.7.
+
+The ``choices_as_values`` option was introduced to ensure backward compatibility with the
```

Comment:
@javiereguiluz What's the official line length? Seems I'm blind as I cannot find anything in the contribution guide. :(